### PR TITLE
Refactor FormulaHelper to read racial resistances

### PR DESCRIPTION
**Pull request by numidium**
FormulaHelper previously applied racial resistances by reading the player's race directly. With this change, the saving throw calculation will be modified by the contents of the player's RaceTemplate.

Addresses the following issue:
https://github.com/Interkarma/daggerfall-unity/issues/2660

### CODE REVIEW

File: Assets/Scripts/Game/Formulas/FormulaHelper.cs [hunk 1]
```diff
@@ -1509,9 +1509,14 @@ public static int SavingThrow(DFCareer.Elements elementType, DFCareer.EffectFlag
                 savingThrow += 25;
 
             savingThrow += biographyMod + modifier;
-            if (elementType == DFCareer.Elements.Frost && target == playerEntity && playerEntity.Race == Races.Nord)
-                savingThrow += 30;
-            else if (elementType == DFCareer.Elements.Magic && target == playerEntity && playerEntity.Race == Races.Breton)
+
+            // Apply racial resistances
+            var raceTemplate = playerEntity.GetLiveRaceTemplate();
```
Comment (KABoissonneault) [inline Assets/Scripts/Game/Formulas/FormulaHelper.cs:1514]: Seems like the check for whether the target is the player entity is missing.

Comment (numidium) [reply Assets/Scripts/Game/Formulas/FormulaHelper.cs]: Yep, fixed that now.

**Review by petchema — APPROVED** [review]

**Review by ajrb — DISMISSED** [review]

**Comment by KABoissonneault** [comment]
I'm not convinced with the changes in this PR. It feels incomplete.
1. High Elf paralysis resistance is still hardcoded rather than going through immunity flags
2. Only racial Resistance is checked, not Immunity, Low Tolerance, or Critical Weakness

I guess the main point is really missing Breton flag for Resistance, so maybe I should let this PR pass. It just feels like a wasted opportunity if we just do the data-driven formula halfway

**Review by KABoissonneault — CHANGES_REQUESTED** [review]
Numidium told me he would rework this PR for completeness

**Comment by numidium** [comment]
Added a new commit that uses RaceTemplate for all saving throw-related flags. See commit description for details.

**Comment by KABoissonneault** [comment]
Seems mostly good. 

1. I think a race that has resistance for both paralysis and the element should apply the effect twice. Not relevant with any of the vanilla race, so I'll keep that as a later improvement.
2. The saving throw for High Elves never goes through. The EntityEffectManager blocks the spell immediately with `IsEntityImmuneToParalysis`. It does not make this PR wrong, just kind of pointless. Perhaps we should remove the race template check in the EntityEffectManager, and let the SavingThrow formula handle it

I'm gonna merge this in the meantime, no critical issues

**Review by KABoissonneault — APPROVED** [review]

**Pull request merged**